src/libmeasurement_kit/ooni/meek_fronted_requests.cpp

```diff
@@ -0,0 +1,88 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include "../common/utils.hpp"
+#include "../ooni/constants.hpp"
+#include <measurement_kit/ooni.hpp>
+
+namespace mk {
+namespace ooni {
+
+using namespace mk::report;
+
+void meek_fronted_requests(std::string input, Settings options,
+                   Callback<Var<report::Entry>> callback,
+                   Var<Reactor> reactor, Var<Logger> logger) {
+    Var<Entry> entry(new Entry);
+
+    std::string expected_body, outer_host, inner_host;
+
+    if (options["expected_body"].empty()) {
+        expected_body = constants::MEEK_SERVER_RESPONSE;
+    } else {
+        expected_body = options["expected_body"];
+    }
+
+    std::list<std::string> outer_inner = split(input, ":");
+    if (outer_inner.size() != 2) {
+        std::string fail_msg = "Couldn't split input: " + input;
+        logger->warn(fail_msg.c_str());
+        (*entry)["failure"] = fail_msg;
```

Comment:
I think we should use existing errors for failures. Here it would probably suffice to use:

```C++
(*entry)["failure"] = ValueError().as_ooni_error();
```